### PR TITLE
Prepare for 050 release

### PR DESCRIPTION
**Pull request by kittylyst**
This adds integration tests, examples and a bit of documentation for the events functionality.

### CODE REVIEW

File: integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    eventBuffer.addEvent(
+        new Event("myEvent", new Attributes().put("key2", "val2"), currentTimeMillis + 1));
+    assertThrows(
+        exceptionClass,
+        () -> eventBatchSender.sendBatch(eventBuffer.createBatch()),
+        exceptionMessage);
+  }
+
+  //
+  // [{\"common\":{\"attributes\":{\"key1\":\"val1\"}},\"metrics\":[{\"name\":\"myEvent\",\"type\":\"count\",\"value\":1.0,\"timestamp\":1557180766612,\"interval.ms\":42,\"attributes\":{\"key2\":\"val2\"}}]}]"
```
Comment (jkwatson) [inline integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java:203]: This comment is not accurate for the Events JSON payload.

File: integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+  private static class EventPayload {
+    private final List<Map<String, Object>> metrics;
```
Comment (jkwatson) [inline integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java:206]: this is weird. why is there a `metrics` member in here.

Comment (jkwatson) [reply integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java:206]: I suspect that the payload isn't being inspected at all by mockserver, due to the `ONLY_MATCHING_FIELDS` option. We should get this payload (and the comment) whipped into shape to make sure that we're actually testing the shape of the payload, and not just the headers, etc.

Comment (kittylyst) [reply integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java:206]: This is exactly the same as the Span and Metrics case.

Comment (jkwatson) [reply integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java:206]: It's not the same. In those cases, the payload structure matches the JSON (`common` + `metrics/spans`). This payload structure is different.

Comment (jkwatson) [reply integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java:206]: I think when lombok got removed, the lack of getters on these little classes made it so we stopped verifying the payload structure. We need that back.

Comment (jkwatson) [reply integration_test/src/test/java/com/newrelic/telemetry/EventApiIntegrationTest.java:206]: I put in a PR to fix this for metrics/spans.  https://github.com/newrelic/newrelic-telemetry-sdk-java/pull/152

**Pull request merged**